### PR TITLE
build: add biobear recipe

### PR DESCRIPTION
**Pull request by tshauck**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

Not sure the ramifications of "Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md)." ... I'm using maturin which builds a rust library that I think would fall under this?

### CODE REVIEW

**Comment by tshauck** [comment]
@conda-forge/help-python-c I think this is ready for review please. I'm tagging `help-python-c` though it actually rust, I thought there may be overlap. Thanks!

**Comment by tshauck** [comment]
Opps... think I actually needed @conda-forge-admin, please ping team @conda-forge/help-python-c

File: recipes/biobear/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  build:
+    - python                                 # [build_platform != target_platform]
+    - {{ compiler('c') }}
+    - {{ compiler('rust') }}
+    - maturin >=1.1.0,<2.0.0                 # [build_platform != target_platform]
+
```
Comment (ocefpaf) [inline recipes/biobear/meta.yaml:22]:
```suggestion
```

File: recipes/biobear/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+build:
+  script: {{ PYTHON }} -m pip install . -vv
+  number: 0
+
+requirements:
+  build:
+    - python                                 # [build_platform != target_platform]
+    - {{ compiler('c') }}
+    - {{ compiler('rust') }}
```
Comment (ocefpaf) [inline recipes/biobear/meta.yaml:20]:
```suggestion
    - {{ compiler('rust') }}  # [build_platform != target_platform]
```

File: recipes/biobear/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+  host:
+    - python
+    - maturin >=1.1.0,<2.0.0
+    - pip
+    - xz                                     # [linux]
+
+  run:
+    - python
+    - pyarrow >=12,<13
+
```
Comment (ocefpaf) [inline recipes/biobear/meta.yaml:32]:
```suggestion
```

File: recipes/biobear/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    - python                                 # [build_platform != target_platform]
+    - {{ compiler('c') }}
+    - {{ compiler('rust') }}
+    - maturin >=1.1.0,<2.0.0                 # [build_platform != target_platform]
+
+  host:
+    - python
+    - maturin >=1.1.0,<2.0.0
+    - pip
+    - xz                                     # [linux]
```
Comment (ocefpaf) [inline recipes/biobear/meta.yaml:27]: Why is this dependency Linux only? Any problems on macOS/Windows that we can help?

Comment (tshauck) [reply recipes/biobear/meta.yaml:26]: Hmmm... it only seemed to be needed to compile on linux, but perhaps I should make it for all platforms?

File: recipes/biobear/meta.yaml (hunk 2 above)
Comment (tshauck) [reply recipes/biobear/meta.yaml]: I committed this suggestion, but it seems to result in an error that cargo is no longer found...

```
Processing $SRC_DIR
  Added file://$SRC_DIR to build tracker '/private/tmp/pip-build-tracker-l58jdo74'
  Created temporary directory: /private/tmp/pip-modern-metadata-rrfepg6q
  Preparing metadata (pyproject.toml): started
  Running command Preparing metadata (pyproject.toml)
  Checking for Rust toolchain....

  Cargo, the Rust package manager, is not installed or is not on PATH.
  This package requires Rust and Cargo to compile extensions. Install it through
  the system's package manager or via https://rustup.rs/

  error: subprocess-exited-with-error
  
  × Preparing metadata (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> See above for output.
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  full command: /Users/runner/mambaforge/conda-bld/biobear_1689783999438/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/bin/python /Users/runner/mambaforge/conda-bld/biobear_1689783999438/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_p/lib/python3.8/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py prepare_metadata_for_build_wheel /tmp/tmpwkixtdki
  cwd: /Users/runner/mambaforge/conda-bld/biobear_1689783999438/work
  Preparing metadata (pyproject.toml): finished with status 'error'
error: metadata-generation-failed

```

**Comment by tshauck** [comment]
Thanks for the review @ocefpaf ... re-requesting review given the changes / question. Thanks again!

File: recipes/biobear/LICENSE [hunk 5]
```diff
@@ -0,0 +1,20 @@
+Copyright (c) 2023 Trent Hauck
```
Comment (ocefpaf) [inline recipes/biobear/LICENSE:1]: The license is in the tarball. You can delete this file.

Comment (tshauck) [reply recipes/biobear/LICENSE]: done in [be768f7](https://github.com/conda-forge/staged-recipes/pull/23350/commits/be768f77a4a4acccb2f31ce32cf536f5c9857c60)

**Comment by tshauck** [comment]
Thanks for the help! @ocefpaf

**Pull request merged**